packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1093,26 +1118,29 @@ export class ContainerRuntime
 
 		const featureGatesForTelemetry: Record<string, boolean | number | undefined> = {};
 
+		// Make sure we've got all the options including internal ones (even though we have to cast back to IContainerRuntimeOptions below)
+		const internalRuntimeOptions: Readonly<Required<IContainerRuntimeOptionsInternal>> = {
+			summaryOptions,
+			gcOptions,
+			loadSequenceNumberVerification,
+			flushMode,
+			compressionOptions,
+			maxBatchSizeInBytes,
+			chunkSizeInBytes,
+			// Requires<> drops undefined from IdCompressorType
+			enableRuntimeIdCompressor: enableRuntimeIdCompressor as "on" | "delayed",
+			enableGroupedBatching,
+			explicitSchemaControl,
+		};
+
 		const runtime = new containerRuntimeCtor(
 			context,
 			registry,
 			metadata,
 			electedSummarizerData,
 			chunks ?? [],
 			aliases ?? [],
-			{
-				summaryOptions,
-				gcOptions,
-				loadSequenceNumberVerification,
-				flushMode,
-				compressionOptions,
-				maxBatchSizeInBytes,
-				chunkSizeInBytes,
-				// Requires<> drops undefined from IdCompressorType
-				enableRuntimeIdCompressor: enableRuntimeIdCompressor as "on" | "delayed",
-				enableGroupedBatching,
-				explicitSchemaControl,
-			},
+			internalRuntimeOptions as Readonly<Required<IContainerRuntimeOptions>>,
```

Comment:
is this cast really necessary? could it be `satisfies` if it is?

Reply:
I did play around with `satisfies`, didn't come up with anything better than this, unfortunately.  I can use `satisfies` to inline this (the way it was) but then still needed to cast, and it was more type goo than I liked to see inlined.

Reply:
Oh, this cast isn't necessary at all.  Thanks.

Reply:
Probably not a bug actually (according to Copilot)
![image](https://github.com/user-attachments/assets/1377d45e-ddf2-4652-8a76-abbc38d96527)